src/Metadata/Driver/DocBlockDriver/DocBlockTypeResolver.php

```diff
@@ -408,15 +409,15 @@ private function getPhpstanType(\ReflectionClass $declaringClass, string $typeHi
         $self = $this;
 
         foreach ($phpDocNode->children as $node) {
-            if ($node instanceof PhpDocTagNode && '@phpstan-type' === $node->name) {
-                $phpstanType = (string) $node->value;
+            if ($node instanceof PhpDocTagNode && $node->value instanceof TypeAliasTagValueNode) {
+                $phpstanType = $node->value->__toString();
                 preg_match_all(self::PHPSTAN_ARRAY_SHAPE, $phpstanType, $foundPhpstanArray);
-                if (isset($foundPhpstanArray[1][0]) && $foundPhpstanArray[1][0] === $typeHint) {
+                if (isset($foundPhpstanArray[0][0]) && $node->value->alias === $typeHint) {
```

Comment:
Just thinking:
It looks like `$node->value->alias === $typeHint` can be moved to condition in line 412. It looks like we can also use  `preg_match` instead of `preg_match_all`. 🤔 WDYT?

Reply:
Yes, done.